libcxx/include/__config

```diff
@@ -283,9 +283,20 @@
 // - `_LIBCPP_ASSERT_COMPATIBLE_ALLOCATOR` -- checks any operations that exchange nodes between containers to make sure
 //   the containers have compatible allocators.
 //
+// - `_LIBCPP_ASSERT_ARGUMENT_WITHIN_DOMAIN` -- checks that the given argument is within the domain of valid arguments
```

Comment:
This category is also added by https://github.com/llvm/llvm-project/pull/77183 (I made these sibling patches to make managing the patches a little easier for me).